### PR TITLE
[MIG] migrate business_requirement_deliverable_crm from 10.0 to 12.0

### PR DESCRIPTION
**Pull request by ruter-lyu**
@elicoidal

### CODE REVIEW

File: business_requirement_deliverable_crm/models/crm.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+
+class CrmLead(models.Model):
+    _inherit = "crm.lead"
+
+    project_id = fields.Many2one(
+        comodel_name='project.project',
+        string='Project',
+        ondelete='set null',
+    )
+    resource_cost_total = fields.Float(
```
Comment (elicoidal) [inline business_requirement_deliverable_crm/models/crm.py:14]: should be deliverable not resource
(might require migration script though)

File: business_requirement_deliverable_crm/models/crm.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+        compute='_compute_resource_cost_total',
+        string='Total Revenue from BR'
+    )
+
+    @api.multi
+    def _compute_resource_cost_total(self):
```
Comment (elicoidal) [inline business_requirement_deliverable_crm/models/crm.py:20]: deliverable

File: business_requirement_deliverable_crm/views/crm_view.xml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+            'active_test': False,
+        }</field>
+    </record>
+    <record id="crm_case_form_view_oppor" model="ir.ui.view">
+        <field name="name">crm.lead.form</field>
+        <field name="model">crm.lead</field>
+        <field name="inherit_id" ref="crm.crm_case_form_view_oppor"/>
+        <field name="priority">17</field>
+        <field name="arch" type="xml">
+            <xpath expr="//button[@name='action_set_won_rainbowman']" position='before'>
```
Comment (elicoidal) [inline business_requirement_deliverable_crm/views/crm_view.xml]: improve readability of XML

File: business_requirement_deliverable_crm/wizard/crm_make_sale.py [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+
+    partner_id = fields.Many2one(
+        'res.partner',
+        'Customer',
+        default=default_partner
+    )
+    update_quotation = fields.Boolean('Update existing quotation')
+
+    @api.multi
+    def make_order(self):
```
Comment (elicoidal) [inline business_requirement_deliverable_crm/wizard/crm_make_sale.py:27]: You might need to refactor this one to be able to fit to #300 
We should be able to generate a SQ either from Opp (based onlist of BR in the linked project) or based on the list of selected BR(s)

**Review by elicoidal — CHANGES_REQUESTED** [review]
Please take into account #300 and refactor the module.

File: business_requirement_deliverable_crm/models/crm.py [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+        self.ensure_one()
+        self.resource_cost_total = sum(
+            [br.total_revenue for br in
```
Comment (JoJoJoJoJoJoJo) [inline business_requirement_deliverable_crm/models/crm.py:23]: No need to transfer to list.

File: business_requirement_deliverable_crm/tests/test_wizard_br.py [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+            order_id)
+        self.assertTrue(lines)
+
+    def test_create_sale_order_line(self):
+        res = self.wizard.make_order()
+        order_id = res.get('res_id', False)
+        lines = self.wizard.prepare_sale_order_line(
+            self.crm_lead_16.id,
+            order_id)
+        try:
```
Comment (JoJoJoJoJoJoJo) [inline business_requirement_deliverable_crm/tests/test_wizard_br.py:142]: There shouldn't be try..catch in UT.

**Review by JoJoJoJoJoJoJo — APPROVED** [review]
Technically LGTM

**Comment by pedrobaeza** [comment]
Closing as old and not green

**Pull request closed without merging**